sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -172,55 +173,64 @@ private static ErrorDetails CreateExceptionDetails(Response response, RequestFai
                 .Append("Status: ")
                 .Append(response.Status.ToString(CultureInfo.InvariantCulture));
 
-            if (!string.IsNullOrEmpty(response.ReasonPhrase))
+            if (response.IsError)
```

Comment:
nit: consider refactoring to call helper routines in favor of a big if/else block?

Reply:
Wouldn't we just push the big if/else to the helper routines? Or did you have something specific in mind to refactor?

Reply:
In the push for code clarity in Azure.Core 2.0 we worked to refactor big if/else statements to something like 

```csharp
if (condition)
{
  A();
}
else
{
  B();
}
```

or similar, so it's easy to conceptualize the high-level algorithm at a glance and then drill into further details in the leaf-nodes of the function call tree.  "Programming by intention" I think it is sometimes called?

In this case, it might be a little more clear to do something like:

```csharp
messageBuilder.Append( response.IsError ? GetErrorResponseMessage() : GetSuccessResponseMessage() );
```

or 

```csharp
if (response.IsError)
{
  AppendErrorResponseMessage(messageBuilder);
}
else
{
  AppendSanitizedSuccessResponseMessage(messageBuilder);
}
```

But it's a fine point, so your call in the end.